src/main/java/com/minecolonies/coremod/entity/citizen/EntityCitizen.java

```diff
@@ -399,6 +414,209 @@ public InteractionResult checkAndHandleImportantInteractions(final Player player
         return InteractionResult.SUCCESS;
     }
 
+    /**
+     * Direct interaction actions with a player
+     *
+     * @param player
+     * @param hand
+     * @return interaction result
+     */
+    private InteractionResult directPlayerInteraction(final Player player, final InteractionHand hand)
+    {
+        final ItemStack usedStack = player.getItemInHand(hand);
+
+        if (isInteractionItem(usedStack) && interactionCooldown > 0)
+        {
+            if (!level.isClientSide())
+            {
+                playSound(SoundEvents.VILLAGER_NO, 0.5f, (float) SoundUtils.getRandomPitch(getRandom()));
+                player.sendMessage(new TranslatableComponent("com.minecolonies.coremod.interaction.notnow", this.getCitizenData().getName()).setStyle(Style.EMPTY.withColor(
+                    ChatFormatting.RED)),
+                  player.getUUID());
+            }
+            return null;
+        }
+
+        if (usedStack.getItem() == Items.GOLDEN_APPLE && getCitizenDiseaseHandler().isSick())
+        {
+            usedStack.setCount(usedStack.getCount() - 1);
+            player.setItemInHand(hand, usedStack);
+
+            if (!level.isClientSide())
+            {
+                if (getRandom().nextInt(2) == 0)
+                {
+                    getCitizenDiseaseHandler().cure();
+                }
+
+                playSound(SoundEvents.PLAYER_LEVELUP, 1.0f, (float) SoundUtils.getRandomPitch(getRandom()));
+                Network.getNetwork()
+                  .sendToTrackingEntity(new VanillaParticleMessage(getX(), getY(), getZ(), ParticleTypes.HAPPY_VILLAGER),
+                    this);
+            }
+
+            interactionCooldown = 20 * 60 * 5;
+            return InteractionResult.CONSUME;
+        }
+
+        if (ISFOOD.test(usedStack) && usedStack.getItem() != Items.GOLDEN_APPLE)
+        {
+            if (isBaby())
+            {
+                childFoodInteraction(usedStack, player, hand);
+            }
+            else
+            {
+                eatFoodInteraction(usedStack, player, hand);
+            }
+            return InteractionResult.CONSUME;
+        }
+
+        if (usedStack.getItem() == Items.BOOK && isBaby())
+        {
+            usedStack.setCount(usedStack.getCount() - 1);
```

Comment:
All of these will actually consume the item fully in creative, which is not how creative is usually supposed to work.  I'm away from my PC right now so I can't check the actual method name, but I'm pretty sure there's a `decrement()` method or sth that handles this without emptying the slot in creative.

Reply:
it doesnt consume it in creative, tested with it

Reply:
Ok, I guess that's handled elsewhere then.  But still, why not just use `usedStack.shrink(1);` instead?

Reply:
fixed^^